internal/events/events.go

```diff
@@ -98,10 +98,10 @@ type Event struct {
 }
 
 type Subscription struct {
-	mask   EventType
-	id     int
-	events chan Event
-	mutex  sync.Mutex
+	mask  EventType
+	id    int
+	C     chan Event
```

Comment:
Hmm. In this way expose a writable channel. Perhaps keep `events` as it is, and add a 

```
func (s* Subscription) C() <-chan Event {
return s.events
}
```

?